v0/destinations/hs/HSTransform-v2.js

```diff
@@ -0,0 +1,490 @@
+      // for now it is just HS CRM custom objects however later it can be
+      // refactored to accomodate upcoming endpoint and make necessary changes
+
+      chunk.forEach(ev => {
+        // if source is of rETL
+        if (ev.message.source === "rETL") {
+          identifyResponseList.push({ ...ev.message.body.JSON });
+          metadata.push(ev.metadata);
+        }
+      });
+
+      // CRM_CREATE_CUSTOM_OBJECTS has objectType value
+      // extract objectType from the end of the endpoint
+      const objectType = chunk[0].message.endpoint.split("/").pop();
+
+      batchEventResponse.batchedRequest.endpoint = BATCH_CREATE_CUSTOM_OBJECTS.replace(
+        ":objectType",
+        objectType
+      );
+    }
+
+    batchEventResponse.batchedRequest.body.JSON = {
+      inputs: identifyResponseList
+    };
+
+    if (batchOperation === "create") {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_CREATE_NEW_CONTACT;
+    } else if (batchOperation === "update") {
+      batchEventResponse.batchedRequest.endpoint = BATCH_IDENTIFY_CRM_UPDATE_NEW_CONTACT;
+    }
+
+    batchEventResponse.batchedRequest.headers = message.headers;
+    batchEventResponse.batchedRequest.params = message.params;
+
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata,
+      destination
+    };
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+};
+
+const batchEvents = destEvents => {
+  let batchedResponseList = [];
+  const trackResponseList = [];
+  // create contact chunck
+  const createContactEventsChunk = [];
+  // update contact chunk
+  const updateContactEventsChunk = [];
+  // general indentify event chunk
+  const eventsChunk = [];
+  destEvents.forEach(event => {
+    // handler for track call
+    // track call does not have batch endpoint
+    if (event.message.endpoint.includes("/events/v3/send")) {
+      const { message, metadata, destination } = event;
+      const endpoint = get(message, "endpoint");
+
+      const batchedResponse = defaultBatchRequestConfig();
+      batchedResponse.batchedRequest.headers = message.headers;
+      batchedResponse.batchedRequest.endpoint = endpoint;
+      batchedResponse.batchedRequest.body = message.body;
+      batchedResponse.batchedRequest.params = message.params;
+      batchedResponse.batchedRequest.method =
+        defaultGetRequestConfig.requestMethod;
+      batchedResponse.metadata = [metadata];
+      batchedResponse.destination = destination;
+
+      trackResponseList.push(
+        getSuccessRespEvents(
+          batchedResponse.batchedRequest,
+          batchedResponse.metadata,
+          batchedResponse.destination
+        )
+      );
+    } else if (
+      event.message.endpoint ===
+      "https://api.hubapi.com/crm/v3/objects/contacts"
```

Comment:
What about other custom objects?

Reply:
for custom objects support rETL team is working on